HACKING.md

```diff
@@ -123,7 +123,15 @@ You can replace `daemonset/machine-config-daemon` with any other MCO component.
 If the componenet already has a `--v=` flag set, the command above still adds
 verbose and ignores the previous one already set.
 If you watch the pods, you'll notice your component restarting as well.
-Note this still requires [disabling the CVO](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusterversion.md#disabling-the-cluster-version-operator).
+
+Note this still requires [disabling the CVO](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusterversion.md#disabling-the-cluster-version-operator)
+or setting the needed objects to unmanaged (`deployment/machine-config-operator`) in the CVO.
+In case you're working on the `server`, `controller` or `daemon`, it also requires disabling the operator
+since the daemonsets and deployments will be overwritten:
```

Comment:
But that's why the current `make deploy-` model bounces (scales down+up) the operator - because we don't wait to wait for it to notice the images change.  (Though the slowness there must really be a bug right?)

I think this text is incorrect - we shouldn't recommend disabling the operator when developing things as some functionality depends on it...my recent osImageURL work depended on patching both the operator and daemon for example.

Reply:
I'm fine with that indeed, we need to remove that wording from the section below as well though and since i.e. the daemon isn't CVO managed there's no real way to edit it without the operator replacing it with its own pristine copy

Reply:
> the daemon isn't CVO managed

The daemon is managed by the operator.

> there's no real way to edit it without the operator replacing it with its own pristine copy

That's not true; the current `make deploy-` scripts have been doing this just fine right?

Again I think the problem is some sort of race condition where even changing `images.json` the operator doesn't notice the change right away and tries to change the deployment back based on its old cached copy.

Reply:
> > there's no real way to edit it without the operator replacing it with its own pristine copy
> 
> That's not true; the current `make deploy-` scripts have been doing this just fine right?

I need to dig deeper but I think, from my testing, the current `make deploy-*` works just because you also patch the images and the operator re-syncing still result in the new built image but if you try editing something else (with the overrides) the daemon will be replaced again

Reply:
Sorry to jump in, but I'm confused by what you wrote @runcom I dont currently use the hack/* tools but when I work on MC*, I only scale down the CVO, I've never had to do anything to the MCO. Once I scale, my changes to each image "stick" and I have no problems.

Reply:
yeah, I was talking about moving away from scaling the CVO to zero and just set some objects to unmanaged!

Reply:
Gotcha ty!